lib/db/main.js

```diff
@@ -203,18 +203,42 @@ const remove = (
 //  mimeType
 // params
 //  graphUri
-const exportData = (conn, database, options = {}, params = {}) => {
+const exportData = (
+  conn,
+  database,
+  options = {},
+  params = {},
+  additionalHandlers = {}
+) => {
   const headers = conn.headers();
   headers.set('Accept', options.mimetype || 'application/ld+json');
 
   const queryParams = {
     'graph-uri': params.graphUri || 'tag:stardog:api:context:all',
   };
+
+  if (params.compression) {
+    queryParams['server-side'] = true; // required for using compression
+    queryParams.compression = params.compression;
+  }
+
   const suffix = `export?${qs.stringify(queryParams)}`;
 
-  return fetch(conn.request(database, suffix), {
+  const fetchPromise = fetch(conn.request(database, suffix), {
     headers,
-  }).then(httpBody);
+  });
+
+  return fetchPromise.then(res => {
+    const shouldContinue = !additionalHandlers.onResponseStart
+      ? true
+      : additionalHandlers.onResponseStart(res);
```

Comment:
Do you think we'll ever need to handle a case where `onResponseStart` is asynchronous?

Also, not a big deal, but I find `const x = condition ? ifTrue : ifFalse` to be more readable than `const x = !condition ? ifFalse : ifTrue`

Reply:
Maybe (in both cases), but I just used the same API/code as is currently used in `lib/query/main.js`. Don't really want to change in either case unless you'd prefer I also make changes there. Doesn't seem too pressing to me, since this code is basically "stop-gap" code until we get stardog.js 2.0.

Reply:
Sure, I'm fine with that.

Reply:
https://github.com/stardog-union/stardog.js/blob/v2/lib/query/main.js#L32 is the code I meant, incidentally.